tool/tctl/common/resource_command.go

```diff
@@ -215,33 +198,38 @@ func (u *ResourceCommand) Create(client *auth.TunClient) error {
 		case "":
 			return trace.BadParameter("missing resource kind")
 		default:
-			return trace.BadParameter("%q is not supported", raw.Kind)
+			// customized creation:
+			if u.Impl != nil && u.Impl.Create != nil {
```

Comment:
I think using interfaces would be more elegant here as you won't need to check this for Nil, simply provide default interface

Reply:
but how would you customize just one method? interface would force to you create a bunch of do-nothing stubs with "ErrorNotImpl" or something... so this would be trading inconvenience in one place (here) for multiple inconveniences down the road. basically I went with Roma's suggestion here. The question of idiomatic golang's version of "subclass & override one method" is still in the air for me...